modes/company/evil-collection-company.el

```diff
@@ -62,9 +62,10 @@ be set through custom or before evil-collection loads."
 (defun evil-collection-company-supported-p (command &rest _)
   "Return non-nil if `evil-state' is in supported states."
   (cond
-    ((eq command 'prefix)
-     (memq evil-state evil-collection-company-supported-states))
-    (t t)))
+   ((not (bound-and-true-p evil-mode)) t)
+   ((eq command 'prefix)
+    (and (bound-and-true-p evil-mode) (memq evil-state evil-collection-company-supported-states)))
```

Comment:
You don't need this second bound-and-true check.

Reply:
I left this in for safe measure. I don't think an extra check will cause much computing power, but I will remove it for your preference.

Reply:
If you left it in, it defeats the purpose of adding the original clause, whose purpose was to simplify the conditional logic.